cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -241,26 +258,13 @@ public void deleteNamespace(Id.Namespace namespaceId) throws DatasetManagementEx
     clientCache.getUnchecked(namespaceId).deleteNamespace();
   }
 
-  private void addModule(Id.DatasetModule moduleId, Class<?> typeClass) throws DatasetManagementException {
-    try {
-      File tempFile = File.createTempFile(typeClass.getName(), ".jar");
-      try {
-        Location tempJarPath = createDeploymentJar(typeClass, new LocalLocationFactory().create(tempFile.toURI()));
-        clientCache.getUnchecked(moduleId.getNamespace()).addModule(moduleId.getId(), typeClass.getName(), tempJarPath);
-      } finally {
-        tempFile.delete();
-      }
-    } catch (IOException e) {
-      String msg = String.format("Could not create jar for deploying dataset module %s with main class %s",
-                                 moduleId, typeClass.getName());
-      LOG.error(msg, e);
-      throw new DatasetManagementException(msg, e);
-    }
-  }
-
-  private static Location createDeploymentJar(Class<?> clz, Location destination) throws IOException {
-    Location tempBundle = destination.getTempFile(".jar");
+  private Location createDeploymentJar(Class<?> clz) throws IOException {
+    File tempDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
+                            cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    tempDir.mkdirs();
```

Comment:
for the usage here, we don't need to, since the following line is to create a temp file inside the tempDir, which will fail and throwing IOException (which we want), if the tempDir is not a directory.